### PR TITLE
esp32: upgrade to esp-idf v5

### PR DESCRIPTION
**Pull request by jkralik**
@cstevens - this is a new updated build for esp32 - just follow steps in https://github.com/iotivity/iotivity-lite/blob/jkralik/feature/upgradeEsp32/port/esp32/README.md

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -98,4 +98,28 @@ jobs:
...
+          git submodule update --init
+          cd ./port/esp32
+          git clone --recursive -b release/v5.0 https://github.com/espressif/esp-idf.git
+          ./esp-idf/install.sh
+          . ./esp-idf/export.sh
+          idf.py set-target esp32
+          ( cd esp-idf/components/mbedtls/mbedtls && patch -p1 < ../../../../../../patches/01-ocf-x509san-anon-psk.patch)
+          ( cd esp-idf/components/mbedtls/mbedtls && patch -p1 < ../../../../patches/mbedtls/02-ocf-mbedtls-config.patch)
+          ( cd esp-idf/components/lwip/lwip && find ../../../../patches/lwip/ -type f -name '*.patch' -exec patch -p1 -i {} \; )
+          idf.py build
```
Comment (Danielius1922) [inline .github/workflows/build.yml:125]: This tests compilation, but isn't it possible to also run the unit tests?

Comment (jkralik) [reply .github/workflows/build.yml:125]: pls could you explain the unit tests for esp ?

Comment (jkralik) [reply .github/workflows/build.yml:125]: To run esp32 unit tests you need to have integrated esp32 HW in some environment

Comment (Danielius1922) [reply .github/workflows/build.yml:125]: Isn't some sort of emulator available? I've found this - https://github.com/mireq/esp32-simulator - but I have no idea how much work would be required to make it work.

File: port/esp32/adapter/src/storage.c [hunk 2]
```diff
@@ -37,30 +40,38 @@ oc_storage_config(const char *store)
...
   store_path[store_path_len] = '\0';
   path_set = true;
-
   return 0;
 }
 
 long
 oc_storage_read(const char *store, uint8_t *buf, size_t size)
 {
-  APP_DBG("oc_storage_read: %s", store);
```
Comment (Danielius1922) [inline port/esp32/adapter/src/storage.c:50]: Are the `APP_` still used anywhere?

File: .github/workflows/build.yml (hunk 1 above)
Comment (jkralik) [reply .github/workflows/build.yml:125]: I tried one year ago and it is not working. For now I don't want to integrate emulator. Maybe in future we will deploy it directly cloud server to esp32 and test it with hub tests.

File: port/esp32/main/CMakeLists.txt [hunk 3]
```diff
@@ -73,7 +73,7 @@ if (CONFIG_TCP)
 endif()
 
 if (CONFIG_CLOUD)
-  add_definitions(-DOC_CLOUD -DOC_DYNAMIC_ALLOCATION -DOC_TCP -DOC_IPV4)
```
[the remark this answers is not in the document]
Comment (jkralik) [reply port/esp32/main/CMakeLists.txt:76]: With the newest iotivity-lite this dependence is not necessary.

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**